File: src/stubs/camera-api/CameraApi.ts
import { Aperture } from "./Aperture";
import { ApiError } from "./ApiError";
import { Camera } from "./Camera";
import { CameraBrowser, EventCallback } from "./CameraBrowser";
import { CameraFile } from "./CameraFile";
import { CameraProperty } from "./CameraProperty";
import { Directory } from "./Directory";
import { ExposureCompensation } from "./ExposureCompensation";
import { Flag } from "./Flag";
import { ImageQuality } from "./ImageQuality";
import { ISOSensitivity } from "./ISOSensitivity";
import { ObjectEvent } from "./ObjectEvent";
import { Option } from "./Option";
import { OutputDevice } from "./OutputDevice";
import { ShutterSpeed } from "./ShutterSpeed";
import { StateEvent } from "./StateEvent";
import { TimeZone } from "./TimeZone";
import { Volume } from "./Volume";

export * from "./Aperture";
export * from "./ApiError";
export * from "./Camera";
export * from "./CameraBrowser";
export * from "./CameraFile";
export * from "./CameraProperty";
export * from "./Directory";
export * from "./ExposureCompensation";
export * from "./Flag";
export * from "./ImageQuality";
export * from "./ISOSensitivity";
export * from "./ObjectEvent";
export * from "./Option";
export * from "./OutputDevice";
export * from "./ShutterSpeed";
export * from "./StateEvent";
export * from "./TimeZone";
export * from "./Volume";

let STUB = 1;
/**
 * @external toPrimitive
 * @type {Symbol}
 */
STUB = 1;

/**
 * Watch for camera events
 * @param {int} [timeout=1000]
 * @return {Function} stop
 */
export const watchCameras = (timeout = 1000): (() => void) => {
    throw new Error("Not implemented - stub only.");
};
/**
 * Global CameraBrowser instance
 * @type {CameraBrowser}
 */
export const cameraBrowser = new CameraBrowser();

/**
 * @module CameraApi
 */
const CameraApi = {
    Aperture,
    ApiError,
    Camera,
    CameraBrowser,
    CameraFile,
    CameraProperty,
    Directory,
    ExposureCompensation,
    Flag,
    ImageQuality,
    ISOSensitivity,
    ObjectEvent,
    Option,
    OutputDevice,
    ShutterSpeed,
    StateEvent,
    TimeZone,
    Volume,
    /**
     * Global CameraBrowser instance
     * @type {CameraBrowser}
     */
    cameraBrowser,
    /**
     * Watch for camera events
     * @param {int} [timeout=1000]
     * @return {Function} stop
     */
    watchCameras
}
export default CameraApi;

File: src/library/camera-api.cc
#include <napi.h>
#include "aperture.h"
#include "api-error.h"
#include "camera-browser.h"
#include "camera-file.h"
#include "camera-property.h"
#include "exposure-compensation.h"
#include "flag.h"
#include "object-event.h"
#include "option.h"
#include "shutter-speed.h"
#include "state-event.h"
#include "volume.h"
#include "directory.h"
#include "image-quality.h"
#include "iso-sensitivity.h"
#include "time-zone.h"
#include "output-device.h"
#include "file-format.h"

namespace CameraApi {
    Napi::Object RegisterModule(Napi::Env env, Napi::Object exports) {
        Napi::HandleScope scope(env);
        Aperture::Init(env, exports);
        ApiError::Init(env, exports);
        CameraBrowserWrap::Init(env, exports);
        CameraFile::Init(env, exports);
        CameraProperty::Init(env, exports);
        CameraWrap::Init(env, exports);
        Directory::Init(env, exports);
        ExposureCompensation::Init(env, exports);
        FileFormat::Init(env, exports);
        Flag::Init(env, exports);
        ImageQuality::Init(env, exports);
        ISOSensitivity::Init(env, exports);
        ObjectEvent::Init(env, exports);
        Option::Init(env, exports);
        OutputDevice::Init(env, exports);
        ShutterSpeed::Init(env, exports);
        StateEvent::Init(env, exports);
        TimeZone::Init(env, exports);
        Volume::Init(env, exports);
        return exports;
    }

    NODE_API_MODULE(
        NODE_GYP_MODULE_NAME, RegisterModule
    )
}
